### PR TITLE
chore: remove deprecated tags fields

### PR DESCRIPTION
**Pull request by alexkim205**
## Problem

Error warning that deprecated fields were being used were spamming in logs. This PR contains migrations that remove this field that hasn't been in use since the [new tagging model was released](https://github.com/PostHog/posthog/pull/8825). 

## Changes

- [ ] Remove deprecated tags fields

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Tests with tags pass, and smoke tests creating, adding, updating, deleting tags on tag-compatible objects all work via app.

Error logs no longer show up

### CODE REVIEW

**Review by neilkakkar — COMMENTED** [review]
Is this a safe operation to do? Given that we query these fields, and the logs are not yet empty already, this sounds risky.

For example: https://github.com/PostHog/posthog/blob/master/ee/api/ee_event_definition.py#L20 . 

Instead, before trying to delete these at all, I'd remove all usages such that the warning log is never called, and then consider deleting these.

Aside: This should go away too: https://github.com/PostHog/posthog/blob/master/posthog/models/activity_logging/activity_log.py#L112

**Comment by alexkim205** [comment]
> Is this a safe operation to do? Given that we query these fields, and the logs are not yet empty already, this sounds risky.
> 
> For example: [https://github.com/PostHog/posthog/blob/master/ee/api/ee_event_definition.py#L20](https://github.com/PostHog/posthog/blob/master/ee/api/ee_event_definition.py?rgh-link-date=2022-06-13T14%3A34%3A18Z#L20) .

I think it should be safe. If you look at the `TaggedItemSerializerMixin`, it intercepts any `tags` field serializations and replaces it by fetching the related `tagged_items` objects. Happy to hop on a quick call if i'm missing anything here.

I am a bit worried that dropping the column here will leave the app in an unusable state (as we've seen a number of times in the past). Ideally I'd want to suppress these warning logs but can't seem to do that easily without forking `django_deprecate_fields`.

> Instead, before trying to delete these at all, I'd remove all usages such that the warning log is never called, and then consider deleting these.
> 
> Aside: This should go away too: [https://github.com/PostHog/posthog/blob/master/posthog/models/activity_logging/activity_log.py#L112](https://github.com/PostHog/posthog/blob/master/posthog/models/activity_logging/activity_log.py?rgh-link-date=2022-06-13T14%3A34%3A18Z#L112)

Ooh good catch

File: ee/migrations/0013_remove_deprecated_tags.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    dependencies = [
+        ("ee", "0012_migrate_tags_v2"),
+    ]
+
+    operations = [
+        migrations.RemoveField(model_name="enterpriseeventdefinition", name="deprecated_tags",),
+        migrations.RemoveField(model_name="enterpriseeventdefinition", name="tags",),
+        migrations.RemoveField(model_name="enterprisepropertydefinition", name="deprecated_tags",),
+        migrations.RemoveField(model_name="enterprisepropertydefinition", name="tags",),
```
Comment (Twixes) [inline ee/migrations/0013_remove_deprecated_tags.py:16]: I don't think we can do this – we still have a [policy of keeping deprecated fields in](https://posthog.com/handbook/engineering/databases/schema-changes#do-not-delete-or-rename-django-models-and-fields). AFAIK this hasn't been made safe yet (particularly for self-hosted deployments which can jump by a few versions).

**Review by Twixes — DISMISSED** [review]

**Comment by hazzadous** [comment]
> I think it should be safe. If you look at the TaggedItemSerializerMixin, it intercepts any tags field serializations and replaces it by fetching the related tagged_items objects. Happy to hop on a quick call if i'm missing anything here.

I'm almost certain that there will be a query being fired off with a `select <all the fields>` somewhere. That it's [logging](https://github.com/3YOURMIND/django-deprecate-fields/blob/master/django_deprecate_fields/deprecate_field.py#L31) I believe suggests that it is being retrieved, and therefore removing this will result in a period where the old version is trying to get it and failing.

That it passes tests suggests that it isn't actually needed. One option would be to remove the field as you've done, but update the migration to [only modify the state](https://docs.djangoproject.com/en/3.2/ref/migration-operations/#separatedatabaseandstate) and leave the data in the database, but remove from code. Not ideal but...

Of course, we could also just not log the deprecation warning as well, change the name of the field in code, use db_column to specify the original. That might be the more sensible thing to do.

**Comment by alexkim205** [comment]
Thanks everyone for the input. I decided to go w/ Harry's last suggestion and changed:

- [ ] Rename "tags" to "deprecated_tags_v2" with db_column remaining as "tags"
- [ ] Remove/obscure all references to "tags", "deprecated_tags" in code
- [ ] Remove `deprecate_field` so that warning isn't logged.

Note, I think the failing CI test can be ignored since we're using db_column to point to the original column name here

**Comment by guidoiaquinti** [comment]
Any update on this?

**Comment by alexkim205** [comment]
Looks like some merge conflicts popped up. I'll resolve them but this should be good for a second round of reviews!

**Review by guidoiaquinti — APPROVED** [review]

File: ee/migrations/0013_silence_deprecated_tags_warnings.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Generated by Django 3.2.13 on 2022-06-13 17:48
+
+import django.contrib.postgres.fields
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (hazzadous) [inline ee/migrations/0013_silence_deprecated_tags_warnings.py:7]: Checking the SQL for this:

```
$ ./manage.py sqlmigrate ee 0013

🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻
️Environment variable DEBUG is set - PostHog is running in DEVELOPMENT MODE!
Be sure to unset DEBUG if this is supposed to be a PRODUCTION ENVIRONMENT!
🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺


🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻
Skipping async migrations setup. This is unsafe in production!
🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺

AXES: BEGIN LOG
AXES: BEGIN LOG
AXES: Using django-axes version 5.9.0
AXES: Using django-axes version 5.9.0
AXES: blocking by IP only.
AXES: blocking by IP only.
BEGIN;
--
-- Rename field tags on enterpriseeventdefinition to deprecated_tags_v2
--
ALTER TABLE "ee_enterpriseeventdefinition" RENAME COLUMN "tags" TO "deprecated_tags_v2";
--
-- Rename field tags on enterprisepropertydefinition to deprecated_tags_v2
--
ALTER TABLE "ee_enterprisepropertydefinition" RENAME COLUMN "tags" TO "deprecated_tags_v2";
--
-- Alter field deprecated_tags_v2 on enterpriseeventdefinition
--
ALTER TABLE "ee_enterpriseeventdefinition" RENAME COLUMN "deprecated_tags_v2" TO "tags";
--
-- Alter field deprecated_tags_v2 on enterprisepropertydefinition
--
ALTER TABLE "ee_enterprisepropertydefinition" RENAME COLUMN "deprecated_tags_v2" TO "tags";
COMMIT;
```

Bit funny that it renames the column to deprecated_tags_v2, then back to tags, this should be fine though as is in a  transaction.

**Pull request merged**